### DISCUSSION TITLE
Is yggdrasil just an overlay network?

### DISCUSSION THREAD

**Discussion by FoundOne** [yggdrasil-network/yggdrasil-go#1013] [Q&A] [answered]
Is yggdrasil just an overlay network, because I can't find too much about this in the documentation or the website? If the answer is yes, are there plans for low level routing which do not rely on the IP protocol?

**Comment by neilalexander** [comment, marked as answer]
Yggdrasil is currently implemented as an overlay that allows peering over existing IP networks and transports IPv6 traffic because it is the _most convenient_ way for us to test and grow the network globally over existing internet links, and to allow users to use existing IP-based applications over Yggdrasil.

That said, there is nothing about the design itself which requires it to be implemented as an overlay — we could just as easily use physical links and either live at, or replace, the link layer — nor does the design actually require that it transport IPv6 packets either.

Yggmail is a fun example of this: it sends raw QUIC packets directly over Yggdrasil instead of encapsulating them in IP.

**Comment by sdgathman** [comment]
yggdrasil uses TCP.  You do not have to use the public internet for this - anything that has TCP works.  E.g. I use Batman wifi radio mesh which looks like ethernet, so of course TCP (and any other IP protocol) works.  A wifi Pt2pt link can go 250 miles line of sight in dry weather - that supports TCP as well.  TCP has trouble with super low bandwidth (Lora) or super high latency (spacenet), but yggdrasil might work with TCP alternatives designed for those situations.

**Reply by robfacella** [reply]
For a client connection, that's part of the multi-homed outbound peering mentioned in the advanced-peering md, right? 

Or does it just work plug-n-play if all of the b.a.t.m.a.n. mesh was configured correctly?
